src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types/Todo';
+import { User } from './components/types/User';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todosDefault: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const getMaxId = (prevTodos: Todo[]) => {
+  return Math.max(...prevTodos.map(({ id }) => id)) + 1;
+};
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(todosDefault);
+  const [userId, setUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const addTodo = () => {
+    if (title && userId !== 0) {
+      setTodos((prevTodos: Todo[]) => (
+        [...prevTodos, {
+          id: getMaxId(prevTodos),
+          title,
+          completed: false,
+          userId,
+          user: getUser(userId),
+        }]
+      ));
+      setTitle('');
+      setUserId(0);
+    }
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (userId === 0) {
+      setUserError(true);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+        }}
```

Comment:
```suggestion
        onSubmit={addTodo}
```
implement addTodo logic here